components/script/dom/node.rs

```diff
@@ -451,18 +451,25 @@ impl Node {
         cmp & NodeConstants::DOCUMENT_POSITION_PRECEDING != 0
     }
 
-    /// Return all registered mutation observers for this node.
-    /// XXX(ferjm) This should probably be split into two functions,
-    /// `registered_mutation_observers`, which returns an Option or
-    /// an empty slice or something, and doesn't create the rare data,
-    /// and `registered_mutation_observers_mut`, which does lazily
-    /// initialize the raredata.
-    pub fn registered_mutation_observers(&self) -> RefMut<Vec<RegisteredObserver>> {
+    /// Return all registered mutation observers for this node. Lazily initialize the
+    /// raredata if it does not exist.
+    pub fn registered_mutation_observers_mut(&self) -> RefMut<Vec<RegisteredObserver>> {
         RefMut::map(self.ensure_rare_data(), |rare_data| {
             &mut rare_data.mutation_observers
         })
     }
 
+    pub fn registered_mutation_observers(&self) -> Option<Ref<Vec<RegisteredObserver>>> {
+        let rare_data: Ref<_> = self.rare_data.borrow();
+
+        if rare_data.is_none() {
+            return None;
+        }
+        Some(Ref::map(rare_data, |rare_data| {
```

Comment:
The check above can be removed and this line can be changed to:

```suggestion
        Some(Ref::map(rare_data?, |rare_data| {
```

Reply:
The type of `rare_data` is `Ref<Option<_>>`. So we cannot do `rare_data?`:
```
error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
   --> components/script/dom/node.rs:465:23
    |
465 |         Some(Ref::map(rare_data?, |rare_data| {
    |                       ^^^^^^^^^^ the `?` operator cannot be applied to type `std::cell::Ref<'_, std::option::Option<std::boxed::Box<dom::raredata::NodeRareData>>>`

```

Reply:
I tried a few other things, but dealing with the `Ref<_>` is annoying.

Reply:
How about `(*rare_data)?` ?

Reply:
Yeah, it tends to inhibit some common patterns. I wouldn't spend any more time trying to clean it up.